stan/math/opencl/kernels/matrix_multiply.hpp

```diff
@@ -99,7 +120,10 @@ static const char* matrix_multiply_kernel_code = STRINGIFY(
       // save the values
       for (int w = 0; w < WORK_PER_THREAD; w++) {
         // each thread saves WORK_PER_THREAD values
-        C[(j + w * THREAD_BLOCK_SIZE_COL) * M + i] = acc[w];
+        // padded threads should not access C
```

Comment:
is it padded threads? or that threads shouldn't access C if they'd be accessing something unpadded? If it was the former I might expect this check to not use `w` and take place outside of the immediate for loop. But probably you will have threads that are part padded, part not right?

Reply:
Yeah this is due to threads that part padded part not.

Reply:
Would you mind also modifying the comment to say something like that? Thanks!

Reply:
Added, sorry for jumping the gun on clicking resolve on this one.